packages/prettier-plugin-java/src/printers/classes.ts

```diff
@@ -330,10 +330,13 @@ export class ClassesPrettierVisitor extends BaseCstPrettierPrinter {
           ctx.variableInitializer![0].children.expression![0].children
             .ternaryExpression[0].children.QuestionMark !== undefined)
       ) {
-        return rejectAndJoin(" ", [
-          variableDeclaratorId,
-          ctx.Equals[0],
-          variableInitializer
+        const groupId = Symbol("assignment");
+        return group([
+          group(variableDeclaratorId),
+          " =",
```

Comment:
I think that with this implementation, some (oddly placed) comment might disapear. For instance, I think that in one of these two statements, the comment will disapear, as it is attached to the ctx.Equals[0] symbol.

`private Map<Integer, String> genericVariable4 /* comment */= new HashMap<Integer, String>();` or `private Map<Integer, String> genericVariable4 =/* comment */ new HashMap<Integer, String>();`

Can you try to add this test case and check if it is the case ? If the comment disappears, you can try to replace " =" by `" ", ctx.Equals[0],` here and in the three declarations afterwards